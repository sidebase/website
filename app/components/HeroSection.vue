<template>
  <div class="flex items-center justify-center h-[55vh] mx-4 md:mt-0">
    <div class="flex flex-col">
      <div class="text-center">
        <img src="/Logo.png" class="h-[150px] w-[150px] md:h-[250px] md:w-[250px] mx-auto select-none">
        <h1 class="text-6xl md:text-8xl text-SideBase-green font-bold">
          sidebase
        </h1>
        <h2 class="text-2xl text-gray-400 font-CooperHewitt font-light">
          sidebase is a modern, best-practice, batteries-included <br>
          fullstack-app starter based on <a href="https://v3.nuxtjs.org/" target="_blank" class="text-SideBase-green underline">Nuxt3</a> and <a href="https://www.typescriptlang.org/" target="_blank" class="text-SideBase-green underline">TypeScript.</a>
        </h2>
        <div class="flex items-center gap-3 w-full justify-center mt-8">
          <a href="https://github.com/sidestream-tech/sidebase" target="_blank">
            <Button label="Star on Github" class="p-button-outlined" icon="pi pi-star" icon-pos="left" />
          </a>
          <a class="hidden md:block" href="https://github.com/sidestream-tech/sidebase/generate" target="_blank">
            <Button label="Use starter template" icon="pi pi-flag" icon-pos="left" />
          </a>
        </div>
      </div>
      <div />
    </div>
  </div>
</template>
