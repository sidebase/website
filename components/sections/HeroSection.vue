<script lang="ts">
import { useClipboard } from '@vueuse/core'
import { defineComponent } from 'vue'
const { copy } = useClipboard()

export default defineComponent({
  data() {
    return {
      wasCopied: false,
    }
  },
  methods: {
    copyToClipBoard() {
      copy('npx nuxi@latest init -t community/sidebase')
      this.wasCopied = true
    },
  },
})
</script>

<template>
  <section>
    <div class="mx-auto max-w-screen-xl px-4 py-5 text-center lg:py-8">
      <div class="my-5 mx-auto max-w-sm">
        <a-popover placement="right">
          <template #content>
            <span class="text-gray-200">
              Paste this one liner into your console!
            </span>
          </template>
          <div class="flex">
            <button class="border-sidebase-green-800 bg-sidebase-green-800 inline-flex items-center rounded-l-md border border-r-0 px-3 text-sm text-white" @click="copyToClipBoard">
              {{ wasCopied ? 'Copied!' : 'Copy' }}
            </button>
            <input
              disabled
              type="text"
              class="border-sidebase-green-800 block w-full min-w-0 flex-1 rounded-r-lg border bg-gray-800 p-2.5 text-center text-sm text-white"
              value="npx nuxi@latest init -t community/sidebase"
            >
          </div>
        </a-popover>
      </div>
      <h1 class="mb-4 text-4xl font-extrabold leading-none tracking-tight text-white md:text-5xl lg:text-6xl">
        The Nuxt3 starter for production!
      </h1>
      <p class="mb-8 text-lg font-normal text-gray-400 sm:px-16 lg:text-xl xl:px-48">
        sidebase is a modern, best-practice, batteries-included fullstack-app starter based on Nuxt3 and TypeScript.
      </p>
      <div class="flex flex-col space-y-4 sm:flex-row sm:justify-center sm:space-y-0 sm:space-x-4">
        <a href="https://github.com/sidestream-tech/sidebase" target="_blank" class="group from-sidebase-green-800 relative inline-flex items-center justify-center overflow-hidden rounded-lg bg-gradient-to-br to-pink-500 p-0.5 text-lg text-white hover:text-white focus:outline-none focus:ring-4 focus:ring-purple-800 group-hover:from-purple-500 group-hover:to-pink-500">
          <span class="relative rounded-md px-5 py-2.5 transition-all duration-200 ease-in group-hover:sm:bg-gray-800">
            <i class="fa-brands fa-github mx-1" />
            View on Github
          </span>
        </a>
        <a href="#getting-started" class="group relative inline-flex items-center justify-center overflow-hidden rounded-lg bg-gradient-to-br from-purple-800 to-green-500 p-0.5 text-lg text-white hover:text-white focus:outline-none focus:ring-4 focus:ring-purple-800 group-hover:from-purple-500 group-hover:to-pink-500">
          <span class="relative rounded-md px-5 py-2.5 transition-all duration-200 ease-in group-hover:sm:bg-gray-800">
            <i class="fa fa-rocket mx-1" />
            Getting started
          </span>
        </a>
      </div>
    </div>
  </section>
</template>

<style lang="scss" scoped>
.GradientText {
  text-align: center;

  background: linear-gradient(to right, #81F6D4 20%, #12a87b 40%, #0FCF97 60%, #81F6D4 80%);
  background-size: 200% auto;

  color: #000;
  background-clip: text;
  text-fill-color: transparent;
  -webkit-background-clip: text;
  -webkit-text-fill-color: transparent;

  animation: shine 1s linear infinite;
  @keyframes shine {
    to {
      background-position: 200% center;
    }
  }
}
</style>

<style lang="scss">
.ant-popover-inner {
  border: 1px solid rgb(11 154 113);
  background-color: rgb(31 41 55);
  border-radius: 5px;
}

.ant-popover-arrow-content {
  border: 1px solid rgb(11 154 113);
  background-color: rgb(31 41 55);
}
</style>
