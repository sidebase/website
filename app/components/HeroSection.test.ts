import { describe, expect, test } from 'vitest'
import HeroSection from './HeroSection.vue'
import { render } from '~/tests/testingLibraryVue'

describe('HeroSection', () => {
  test('mounts and is still the same', () => {
    const { html } = render(HeroSection)
    expect(html()).toMatchSnapshot()
  })
})
