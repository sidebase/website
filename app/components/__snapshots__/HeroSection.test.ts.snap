// Vitest Snapshot v1

exports[`HeroSection > mounts and is still the same 1`] = `
"<div class=\\"flex items-center justify-center h-[70vh] mx-4 md:mt-0\\">
  <div class=\\"flex flex-col\\">
    <div class=\\"text-center\\">
      <h1 class=\\"text-6xl md:text-8xl text-SideBase-green font-bold\\"> sidebase </h1>
      <h2 class=\\"text-2xl text-gray-400 font-CooperHewitt font-light\\"> sidebase is a modern, best-practice, batteries-included <br> fullstack-app starter based on <a href=\\"https://v3.nuxtjs.org/\\" target=\\"_blank\\" class=\\"text-SideBase-green underline\\">Nuxt3</a> and <a href=\\"https://www.typescriptlang.org/\\" target=\\"_blank\\" class=\\"text-SideBase-green underline\\">TypeScript.</a></h2>
      <div class=\\"flex items-center gap-3 w-full justify-center mt-8\\"><a href=\\"https://github.com/sidestream-tech/sidebase\\" target=\\"_blank\\"><button class=\\"p-button p-component p-button-outlined\\" type=\\"button\\">
            <!----><span class=\\"pi pi-star p-button-icon p-button-icon-left\\"></span><span class=\\"p-button-label\\">Star on Github</span>
            <!---->
          </button></a><a class=\\"hidden md:block\\" href=\\"https://github.com/sidestream-tech/sidebase/generate\\" target=\\"_blank\\"><button class=\\"p-button p-component\\" type=\\"button\\">
            <!----><span class=\\"pi pi-flag p-button-icon p-button-icon-left\\"></span><span class=\\"p-button-label\\">Use starter template</span>
            <!---->
          </button></a></div>
    </div>
    <div class=\\"hidden lg:block max-w-6xl mx-auto mt-20\\">
      <div id=\\"pv_id_1\\" class=\\"p-carousel p-component p-carousel-horizontal\\">
        <!---->
        <div class=\\"p-carousel-content\\">
          <div class=\\"p-carousel-container\\"><button class=\\"p-carousel-prev p-link\\" type=\\"button\\"><span class=\\"p-carousel-prev-icon pi pi-chevron-left\\"></span></button>
            <div class=\\"p-carousel-items-content\\" style=\\"height: auto;\\">
              <div class=\\"p-carousel-items-container\\">
                <div class=\\"p-carousel-item p-carousel-item-cloned p-carousel-item-start p-carousel-item-end\\">
                  <div class=\\"w-full py-10\\">
                    <div class=\\"text-center py-6 text-xl lg:text-2xl text-SideBase-green\\"> \\"Beautiful work on sidebase!\\" </div>
                    <div class=\\"flex items-center space-x-3 mx-auto w-max\\">
                      <div class=\\"p-avatar p-component p-avatar-image p-avatar-circle p-avatar-lg\\"><img src=\\"https://avatars.githubusercontent.com/u/904724?v=4\\"></div>
                      <div>
                        <h1 class=\\"font-bold text-lg\\">Atinux <a class=\\"font-bold underline text-SideBase-green ml-0.5 pt-2\\" href=\\"https://twitter.com/Atinux\\" target=\\"_blank\\"><i class=\\"pi pi-twitter\\" style=\\"font-size: 1.1rem;\\"></i></a></h1>
                        <p class=\\"text-sm opacity-75\\">CEO of NuxtJS</p>
                      </div>
                    </div>
                  </div>
                </div>
                <div class=\\"p-carousel-item\\">
                  <div class=\\"w-full py-10\\">
                    <div class=\\"text-center py-6 text-xl lg:text-2xl text-SideBase-green\\"> \\"Been following this issue/project for a few weeks now - this is fantastic, and also a project extremely close to my interest!\\" </div>
                    <div class=\\"flex items-center space-x-3 mx-auto w-max\\">
                      <div class=\\"p-avatar p-component p-avatar-image p-avatar-circle p-avatar-lg\\"><img src=\\"https://avatars.githubusercontent.com/u/56732164?v=4\\"></div>
                      <div>
                        <h1 class=\\"font-bold text-lg\\">ineshbose
                          <!--v-if-->
                        </h1>
                        <p class=\\"text-sm opacity-75\\"></p>
                      </div>
                    </div>
                  </div>
                </div>
                <div class=\\"p-carousel-item\\">
                  <div class=\\"w-full py-10\\">
                    <div class=\\"text-center py-6 text-xl lg:text-2xl text-SideBase-green\\"> \\"Just put my start, the template looks awesome!\\" </div>
                    <div class=\\"flex items-center space-x-3 mx-auto w-max\\">
                      <div class=\\"p-avatar p-component p-avatar-image p-avatar-circle p-avatar-lg\\"><img src=\\"https://pbs.twimg.com/profile_images/1364816201646563335/TF8owRfC_400x400.png\\"></div>
                      <div>
                        <h1 class=\\"font-bold text-lg\\">elbrujo <a class=\\"font-bold underline text-SideBase-green ml-0.5 pt-2\\" href=\\"https://twitter.com/elbrujolegal\\" target=\\"_blank\\"><i class=\\"pi pi-twitter\\" style=\\"font-size: 1.1rem;\\"></i></a></h1>
                        <p class=\\"text-sm opacity-75\\"></p>
                      </div>
                    </div>
                  </div>
                </div>
                <div class=\\"p-carousel-item\\">
                  <div class=\\"w-full py-10\\">
                    <div class=\\"text-center py-6 text-xl lg:text-2xl text-SideBase-green\\"> \\"Sidebase looks great. Awesome!\\" </div>
                    <div class=\\"flex items-center space-x-3 mx-auto w-max\\">
                      <div class=\\"p-avatar p-component p-avatar-image p-avatar-circle p-avatar-lg\\"><img src=\\"https://avatars.githubusercontent.com/u/188172?v=4\\"></div>
                      <div>
                        <h1 class=\\"font-bold text-lg\\">ManUtopiK <a class=\\"font-bold underline text-SideBase-green ml-0.5 pt-2\\" href=\\"https://twitter.com/ManUtopiK\\" target=\\"_blank\\"><i class=\\"pi pi-twitter\\" style=\\"font-size: 1.1rem;\\"></i></a></h1>
                        <p class=\\"text-sm opacity-75\\"></p>
                      </div>
                    </div>
                  </div>
                </div>
                <div class=\\"p-carousel-item\\">
                  <div class=\\"w-full py-10\\">
                    <div class=\\"text-center py-6 text-xl lg:text-2xl text-SideBase-green\\"> \\"Thanks for sharing! Nice start to explore @nuxt_js 3.\\" </div>
                    <div class=\\"flex items-center space-x-3 mx-auto w-max\\">
                      <div class=\\"p-avatar p-component p-avatar-image p-avatar-circle p-avatar-lg\\"><img src=\\"https://pbs.twimg.com/profile_images/954370626352971776/DPG_EqzI_400x400.jpg\\"></div>
                      <div>
                        <h1 class=\\"font-bold text-lg\\">anorakio <a class=\\"font-bold underline text-SideBase-green ml-0.5 pt-2\\" href=\\"https://twitter.com/anorakio\\" target=\\"_blank\\"><i class=\\"pi pi-twitter\\" style=\\"font-size: 1.1rem;\\"></i></a></h1>
                        <p class=\\"text-sm opacity-75\\"></p>
                      </div>
                    </div>
                  </div>
                </div>
                <div class=\\"p-carousel-item\\">
                  <div class=\\"w-full py-10\\">
                    <div class=\\"text-center py-6 text-xl lg:text-2xl text-SideBase-green\\"> \\"Beautiful work on sidebase!\\" </div>
                    <div class=\\"flex items-center space-x-3 mx-auto w-max\\">
                      <div class=\\"p-avatar p-component p-avatar-image p-avatar-circle p-avatar-lg\\"><img src=\\"https://avatars.githubusercontent.com/u/904724?v=4\\"></div>
                      <div>
                        <h1 class=\\"font-bold text-lg\\">Atinux <a class=\\"font-bold underline text-SideBase-green ml-0.5 pt-2\\" href=\\"https://twitter.com/Atinux\\" target=\\"_blank\\"><i class=\\"pi pi-twitter\\" style=\\"font-size: 1.1rem;\\"></i></a></h1>
                        <p class=\\"text-sm opacity-75\\">CEO of NuxtJS</p>
                      </div>
                    </div>
                  </div>
                </div>
                <div class=\\"p-carousel-item p-carousel-item-cloned p-carousel-item-active p-carousel-item-start p-carousel-item-end\\">
                  <div class=\\"w-full py-10\\">
                    <div class=\\"text-center py-6 text-xl lg:text-2xl text-SideBase-green\\"> \\"Been following this issue/project for a few weeks now - this is fantastic, and also a project extremely close to my interest!\\" </div>
                    <div class=\\"flex items-center space-x-3 mx-auto w-max\\">
                      <div class=\\"p-avatar p-component p-avatar-image p-avatar-circle p-avatar-lg\\"><img src=\\"https://avatars.githubusercontent.com/u/56732164?v=4\\"></div>
                      <div>
                        <h1 class=\\"font-bold text-lg\\">ineshbose
                          <!--v-if-->
                        </h1>
                        <p class=\\"text-sm opacity-75\\"></p>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div><button class=\\"p-carousel-next p-link\\" type=\\"button\\"><span class=\\"p-carousel-prev-icon pi pi-chevron-right\\"></span></button>
          </div>
          <ul class=\\"p-carousel-indicators p-reset\\">
            <li class=\\"p-carousel-indicator p-highlight\\"><button class=\\"p-link\\" type=\\"button\\"></button></li>
            <li class=\\"p-carousel-indicator\\"><button class=\\"p-link\\" type=\\"button\\"></button></li>
            <li class=\\"p-carousel-indicator\\"><button class=\\"p-link\\" type=\\"button\\"></button></li>
            <li class=\\"p-carousel-indicator\\"><button class=\\"p-link\\" type=\\"button\\"></button></li>
            <li class=\\"p-carousel-indicator\\"><button class=\\"p-link\\" type=\\"button\\"></button></li>
          </ul>
        </div>
        <!---->
      </div>
    </div>
    <div></div>
  </div>
</div>"
`;
